### ISSUE TITLE
Fails with dockerfile containing umlauts (special character)

### ISSUE THREAD

**Issue by shoeper** [hadolint/hadolint#172]
- [x] This is a bug report

### Expected behavior

Run as usual and output results.

### Actual behavior

Having umlauts and possilbly other special characters in the Dockerfile breaks hadolint. hadolint: /dev/stdin: hGetContents: invalid argument (invalid byte sequence)

### Steps to reproduce the behavior

Minimal example: output is
```
docker run --rm -i hadolint/hadolint < Dockerfile.test2
hadolint: /dev/stdin: hGetContents: invalid argument (invalid byte sequence)
```

Dockerfile.test2
```Dockerfile
FROM debian:latest

LABEL maintainer="ö"

RUN echo "Hallo"
```

Running without the `ö` does work as espected. It outputs:
```
/dev/stdin:1 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag
```

```bash
Haskell Dockerfile Linter v1.5.0-13-gc472436
```

**Comment by zemanlx** [comment]
Thanks for raising this interesting issue! 

Seems to be connected to build environment (which can be solved really fast) and busybox's weird handling of utf/unicode (seems non-trivial). I will dig a bit deeper and let you know my findings (and hopefully create a fix).

**Comment by shoeper** [comment]
Thank you very much. If I can assist you somehow, let me know.

(Btw: The issues also exists when binding the file to the container and passing it via command line.)

**Comment by zemanlx** [comment]
Yes, that looks like the only quick solution for me. I am trying it with `LC_ALL=C.UTF-8`

```bash
$ docker images hadolint
REPOSITORY          TAG                        IMAGE ID            CREATED             SIZE
hadolint            v1.5.0-13-gc472436-dirty   23cd9486ca76        54 seconds ago      59MB

$ docker run --rm -i hadolint:v1.5.0-13-gc472436-dirty < Dockerfile.test2
/dev/stdin:1 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag

$ cat Dockerfile.test2                                                   
FROM debian:latest

LABEL maintainer="ö"

RUN echo "Hallo"
```

**Comment by zemanlx** [comment]
I do not know what other options we have. Building Haskell code under alpine is a pain as `stack` and `ghc` are not supported for Alpine. Cirros image looks to have the same issue as busybox. Do you know any other small `glibc` docker image with UTF support?

**Comment by zemanlx** [comment]
Can you try my image `docker pull zemanlx/hadolint:v1.5.0-13-gc472436-dirty` with your Dockerfiles and let me know if it is working?

**Comment by shoeper** [comment]
Yes, it works.

**Comment by lorenzo** [comment]
@zemanlx what did you do to fix it?

**Comment by zemanlx** [comment]
@lorenzo I have created distro image based on Debian (it uses `locale`) and set 

```Dockerfile
ENV LC_ALL=C.UTF-8
```

it needs a bit more polishing and testing, but I think we will end up with two distro images to solve this.

My work is here https://github.com/hadolint/hadolint/compare/master...zemanlx:fix/UTF-input

**Comment by jbergstroem** [comment]
Just ran into this as well. Can confirm that the fix solves it.
